airflow/api_connexion/endpoints/pool_endpoint.py

```diff
@@ -26,18 +30,33 @@ def delete_pool():
     raise NotImplementedError("Not implemented yet.")
 
 
-def get_pool():
+@provide_session
+def get_pool(pool_name, session):
     """
     Get a pool
     """
-    raise NotImplementedError("Not implemented yet.")
+    pool_id = pool_name
+    query = session.query(Pool)
+    pool = query.filter(Pool.pool == pool_id).one_or_none()
+
+    if pool is None:
+        raise NotFound("Pool not found")
+    return pool_schema.dump(pool)
 
 
-def get_pools():
+@provide_session
+def get_pools(session):
     """
     Get all pools
     """
-    raise NotImplementedError("Not implemented yet.")
+    offset = request.args.get(parameters.page_offset, 0)
+    limit = min(request.args.get(parameters.page_limit, 100), 100)
```

Comment:
```suggestion
    limit = min(int(request.args.get(parameters.page_limit, 100)), 100)
```

Reply:
You can add tests to handle pagination too

Reply:
Thanks
Here is the change `983e673`